Directory.Build.props

```diff
@@ -86,7 +86,7 @@
     <MicrosoftGraphVersion>4.36.0</MicrosoftGraphVersion>
     <MicrosoftGraphBetaVersion>4.57.0-preview</MicrosoftGraphBetaVersion>
     <MicrosoftExtensionsHttpVersion>3.1.3</MicrosoftExtensionsHttpVersion>
-    <MicrosoftIdentityAbstractions>5.3.0</MicrosoftIdentityAbstractions>    
+    <MicrosoftIdentityAbstractions>5.4.0-preview3-HeaderProvider</MicrosoftIdentityAbstractions>    
```

Comment:
This PR will not be merged until we align on the interface definition in Microsoft.Identity.Abstractions and release it.

Reply:
It would be a 6.0.0

Reply:
Updated MicrosoftIdentityAbstractions to 6.0.0